src/base/reconstruction.h

```diff
@@ -192,15 +207,19 @@ class Reconstruction {
   // Align the given reconstruction with a set of pre-defined camera positions.
   // Assuming that locations[i] gives the 3D coordinates of the center
   // of projection of the image with name image_names[i].
+  template<bool kEstimateScale = true>
   bool Align(const std::vector<std::string>& image_names,
              const std::vector<Eigen::Vector3d>& locations,
-             const int min_common_images);
+             const int min_common_images, SimilarityTransform3& tform);
 
   // Robust alignment using RANSAC.
+  template <bool kEstimateScale = true>
   bool AlignRobust(const std::vector<std::string>& image_names,
                    const std::vector<Eigen::Vector3d>& locations,
                    const int min_common_images,
-                   const RANSACOptions& ransac_options);
+                   const RANSACOptions& ransac_options,
+                   SimilarityTransform3& tform);
```

Comment:
I can convert the `tform` input to pointer with `nullptr` default if that's more compatible with the existing colmap style.